server.js

```diff
@@ -1,38 +1,140 @@
+  },
+  category: {
+    type: String,
+  },
+  nominee: {
+    type: String,
+  },
+  film: {
+    type: String,
+  },
+  win: {
+    type: Boolean,
+  },
+});
+
+// RESET_DATABASE=true npm run dev
+if (process.env.RESET_DATABASE) {
+  const seedDatabase = async () => {
+    await GoldenGlobes.deleteMany();
+    await goldenGlobesData.forEach((goldenGlobe) => new GoldenGlobes(goldenGlobe).save());
+  };
+  seedDatabase();
+}
+
 const port = process.env.PORT || 8080
 const app = express()
-
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next();
+  } else {
+    res.status(503).json({ error: 'Service unavailable'});
+  }
+})
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get('/', async (req, res) => {
+  const goldenGlobes = await GoldenGlobes.find();
+  res.json(goldenGlobes);
+})
+
+app.get('/year/:year', async (req, res) => {
+  try {
+    const { year } = req.params;
+    const { won, category } = req.query;    
+    const mongoQuery = { year_award: year };
+    
+    if (won) {
+      mongoQuery.win = won
+    } 
+    
+    if (category) {
+      mongoQuery.category = new RegExp(category, 'i');
+    }
+
+    const goldenGlobesYear = await GoldenGlobes.find(mongoQuery);
+    if (goldenGlobesYear.length < 1) {
+      res.status(404).json({ error: 'Data not found' });
+    } else {
+      res.json(goldenGlobesYear);
+    }
+  } catch (err) {
+    res.status(400).json({ error: 'Invalid request'});
+  }
+})
+
+app.get('/category/:category', async (req, res) => {
+  try {
+    const { category } = req.params;
+    const { won } = req.query;    
+    const mongoQuery = { category: category };
+    
+    if (won) {
+      mongoQuery.win = won
+    }
+
+    const goldenGlobesCategory = await GoldenGlobes.find(mongoQuery);
+    if (goldenGlobesCategory.length < 1) {
+      res.status(404).json({ error: 'Data not found' });
+    } else {
+      res.json(goldenGlobesCategory);
+    }
+  } catch (err) {
+    res.status(400).json({ error: 'Invalid request'});
+  }
+})
+
+app.get('/search/nominee', async (req, res) => {
+  try {
+    const { nominee } = req.query;
+    const queryRegex = new RegExp(nominee, 'i');
+    const goldenGlobesNominee = await GoldenGlobes.find({ nominee: queryRegex });
+    if (goldenGlobesNominee.length < 1) {
+      res.status(404).json({ error: 'Nominee not found' });
+    } else {
+      res.json(goldenGlobesNominee);
+    }
+  } catch (err) {
+    res.status(400).json({ error: 'Invalid request'}); 
+  }
+})
+
+app.get('/nominee/:id', async (req, res) => {
+  try {
+    const { id } = req.params;
+    const goldenGlobesNominee = await GoldenGlobes.findOne({ _id: id });
+    
+    if (goldenGlobesNominee.length < 1) {
+      res.status(404).json({ error: 'Nominee not found' });
+    } else {
+      res.json(goldenGlobesNominee);
+    }
```

Comment:
Nice! There's actually a mongo method specifically to search for _id which is `findById`.